homeassistant/components/omnilogic/sensor.py

```diff
@@ -0,0 +1,484 @@
+        self._attrs["hayward_unit_of_measure"] = unit_of_measurement
+        if temp_state is not None:
+            self._state = float(temp_state)
+            self._unit = TEMP_FAHRENHEIT
+
+        return self._state
+
+
+class OmniLogicPumpSpeedSensor(OmnilogicSensor):
+    """Define an OmniLogic Pump Speed Sensor."""
+
+    def __init__(
+        self,
+        coordinator: OmniLogicUpdateCoordinator,
+        state_key: str,
+        name: str,
+        kind: str,
+        item_id: tuple,
+        device_class: str,
+        icon: str,
+        unit: str,
+    ):
+        """Initialize the sensor."""
+        super().__init__(
+            coordinator=coordinator,
+            kind=kind,
+            name=name,
+            device_class=device_class,
+            icon=icon,
+            unit=unit,
+            item_id=item_id,
+            state_key=state_key,
+        )
+
+    @property
+    def state(self):
+        """Return the state for the pump speed sensor."""
+
+        pump_type = PUMP_TYPES.get(
+            self.coordinator.data[self._item_id].get("Filter-Type")
+        )
+        pump_speed = self.coordinator.data[self._item_id].get(self._state_key)
+
+        if pump_type == "VARIABLE":
+            self._unit = PERCENTAGE
+            self._state = pump_speed
+        elif pump_type == "DUAL":
+            if pump_speed == 0:
+                self._state = "off"
+            elif pump_speed == self.coordinator.data[self._item_id].get(
+                "Min-Pump-Speed"
+            ):
+                self._state = "low"
+            elif pump_speed == self.coordinator.data[self._item_id].get(
+                "Max-Pump-Speed"
+            ):
+                self._state = "high"
+        elif pump_type == "SINGLE":
+            if pump_speed == 0:
+                self.state = "off"
+            elif pump_speed == self.coordinator.data[self._item_id].get(
+                "Max-Pump-Speed"
+            ):
+                self._state = "on"
+
+        self._attrs["pump_type"] = pump_type
+
+        return self._state
+
+
+class OmniLogicSaltLevelSensor(OmnilogicSensor):
+    """Define an OmniLogic Salt Level Sensor."""
+
+    def __init__(
+        self,
+        coordinator: OmniLogicUpdateCoordinator,
+        state_key: str,
+        name: str,
+        kind: str,
+        item_id: tuple,
+        device_class: str,
+        icon: str,
+        unit: str,
+    ):
+        """Initialize the sensor."""
+        super().__init__(
+            coordinator=coordinator,
+            kind=kind,
+            name=name,
+            device_class=device_class,
+            icon=icon,
+            unit=unit,
+            item_id=item_id,
+            state_key=state_key,
+        )
+
+    @property
+    def state(self):
+        """Return the state for the salt level sensor."""
+
+        salt_return = self.coordinator.data[self._item_id].get(self._state_key)
+        unit_of_measurement = CONCENTRATION_PARTS_PER_MILLION
```

Comment:
The default unit should be set in the entity settings dict constant.

Reply:
Addressed (for all)